src/main/java/roboy/memory/nodes/Roboy.java

```diff
@@ -0,0 +1,132 @@
+            // Pick first if matches found.
+            if (ids != null && !ids.isEmpty()) {
+                try {
+                    this.roboy = memory.getById(ids.get(0));
+                } catch (InterruptedException | IOException e) {
+                    System.out.println("Unexpected memory error: provided ID not found upon querying. Go the amnesia mode");
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    /**
+     * Method to obtain the name of the Roboy node
+     * @return String name - text containing the name as in the Memory
+     */
+    public String getName() {
+        return (String) roboy.getProperty("name");
+    }
+
+    /**
+     * Method to obtain the specific type relationships of the Roboy node
+     * @return ArrayList<Integer> ids - list containing integer IDs of the nodes
+     * related to the Roboy by specific relationship type as in the Memory
+     */
+    public ArrayList<Integer> getRelationships(Neo4jRelationships type) {
+        return roboy.getRelationship(type.type);
+    }
+
+    /**
+     * Adds a new relation to the Roboy node, updating memory.
+     */
+    public void addInformation(String relationship, String name) {
+        if(!memoryROS) return;
+        ArrayList<Integer> ids = new ArrayList<>();
+        // First check if node with given name exists by a matching query.
+        MemoryNodeModel relatedNode = new MemoryNodeModel(true);
+        relatedNode.setProperty("name", name);
+        //This adds a label type to the memory query depending on the relation.
+        relatedNode.setLabel(determineNodeType(relationship));
+        try {
+            ids = memory.getByQuery(relatedNode);
+        } catch (InterruptedException | IOException e) {
+            System.out.println("Exception while querying memory by template.");
+            e.printStackTrace();
+        }
+        // Pick first from list if multiple matches found.
+        if(ids != null && !ids.isEmpty()) {
+            roboy.setRelationship(relationship, ids.get(0));
+        }
+        // Create new node if match is not found.
+        else {
+            try {
+                int id = memory.create(relatedNode);
+                if(id != 0) { // 0 is default value, returned if Memory response was FAIL.
+                    roboy.setRelationship(relationship, id);
+                }
+            } catch (InterruptedException | IOException e) {
+                System.out.println("Unexpected memory error: creating node for new relation failed.");
+                e.printStackTrace();
+            }
+        }
+        //Update the Roboy node in memory.
+        try{
+            memory.save(roboy);
+        } catch (InterruptedException | IOException e) {
+            System.out.println("Unexpected memory error: updating person information failed.");
+            e.printStackTrace();
+        }
+    }
+
+    private String determineNodeType(String relationship) {
```

Comment:
why is it in Roboy node?

Reply:
This is the method to check for the type of the nodes related to Roboy. I didn't try changing this approach so far as it had been implemented before by Laura. I will ask her about it

Reply:
@Waguramu @l-laura 
from what i understand it's related not just to RoboyNode, but any node related to memory

Reply:
Moved from Roboy and Interlocutor to the Neo4jMemory class as you are completely right